src/moa/moa_controllers/moa_controllers/ack_to_can.py

```diff
@@ -0,0 +1,111 @@
+#/usr/bin/env python3
+from typing import Optional
+import rclpy
+from rclpy.node import Node
+import time
+
+from ackermann_msgs.msg import AckermannDriveStamped
+from ackermann_msgs.msg import AckermannDrive
+from moa_msgs.msg import CANStamped
+from std_msgs.msg import Header
+from builtin_interfaces.msg import Time
+
+import numpy as np
+import zlib
+from math import pi
+
+
+def compress_floats(values:np.ndarray) -> bytes:    
+    # Compute the differences between consecutive values
+    differences = np.diff(values)
+
+    # Compress the differences using zlib
+    compressed = zlib.compress(differences, level=9)
+
+    return compressed
+
+
+def ackermann_to_can_parser(ack_msg: AckermannDriveStamped) -> Optional[CANStamped]: #type for option optionalnull
+        #checks before sending Ackermann
+        if 0 < ack_msg.drive.speed > 120/3.6: #m/s   
+            return None
+        
+        elif 0 < ack_msg.drive.acceleration > 256: #m/s^2
+            return None
+
+        elif 0 < ack_msg.drive.jerk: #m/s^3 
+            #not too fussed about assign 1 byte (minifloat)
+            return None
+        
+        elif -45*pi/180 < ack_msg.drive.steering_angle > 45*pi/180: #radians
+            return None
+        
+        elif ack_msg.drive.steering_angle_velocity: #radians/s
+            return None
+        
+        #format values of Ackermann
+        ackermann_vals = [
+            np.uint8(round(ack_msg.drive.speed)), #(functionally limited to 100-120km/h) integer and round 
+            np.uint8(round(ack_msg.drive.acceleration)), #need to check between 0-256 
+            np.uint8(ack_msg.drive.jerk*1000), #need to check between 0-256 if dont need document that dont need and don't parse
```

Comment:
Also now that I think about it we should leave this in meters since that can easily give us more compression since 28 m/s is 100km/h

Reply:
already in metres. its only the check that is in km/h